sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -150,16 +151,31 @@ void *alignedAlloc(size_t Alignment, size_t Size, const context &Ctxt,
     }
     case alloc::shared: {
       Id = detail::getSyclObjImpl(Dev)->getHandleRef();
+
+      std::vector<pi_usm_mem_properties> Props;
+      Props.reserve(5); // maximum number of elements inserted below
+
       if (PropList.has_property<
               cl::sycl::ext::oneapi::property::usm::device_read_only>()) {
-        pi_usm_mem_properties Props[3] = {PI_MEM_ALLOC_FLAGS,
-                                          PI_MEM_ALLOC_DEVICE_READ_ONLY, 0};
-        Error = Plugin.call_nocheck<PiApiKind::piextUSMSharedAlloc>(
-            &RetVal, C, Id, Props, Size, Alignment);
-      } else {
-        Error = Plugin.call_nocheck<PiApiKind::piextUSMSharedAlloc>(
-            &RetVal, C, Id, nullptr, Size, Alignment);
+        Props.emplace_back(PI_MEM_ALLOC_FLAGS);
+        Props.emplace_back(PI_MEM_ALLOC_DEVICE_READ_ONLY);
+      }
+
+      if (Dev.has_extension("cl_intel_mem_alloc_buffer_location") &&
+          PropList.has_property<cl::sycl::ext::intel::experimental::property::
```

Comment:
How can we track updating this when the buffer_location property graduates out of the experimental namespace?

Reply:
If I interpret the [Experimental Extensions documentation](https://github.com/intel/llvm/blob/b545c60b38da9ba8246a740f88faf80f5bd48379/sycl/doc/extensions/experimental/README.md) correctly, the property would simply be moved to the non-experimental namespace which would naturally enforce an update of this code and the test.